bigtable/autoscaler/autoscaler.py

```diff
@@ -0,0 +1,134 @@
+           up (bool): If true, scale up, otherwise scale down
+    """
+    bigtable_client = bigtable.Client(admin=True)
+    instance = bigtable_client.instance(bigtable_instance)
+    instance.reload()
+
+    cluster = instance.cluster('{}-cluster'.format(bigtable_instance))
+    cluster.reload()
+
+    current_node_count = cluster.serve_nodes
+
+    if current_node_count <= 3 and not up:
+        # Can't downscale lower than 3 nodes
+        return
+
+    if up:
+        strategies_dict = strategies.UPSCALE_STRATEGIES
+    else:
+        strategies_dict = strategies.DOWNSCALE_STRATEGIES
+
+    strategy = strategies_dict['incremental']
+    new_node_count = strategy(cluster.serve_nodes)
+    cluster.serve_nodes = new_node_count
+    cluster.update()
+    print('Scaled from {} up to {} nodes.'.format(
+        current_node_count, new_node_count))
+
+
+def main(
+        bigtable_instance,
+        high_cpu_threshold,
+        low_cpu_threshold,
+        short_sleep,
+        long_sleep):
+    """Main loop runner that autoscales Bigtable.
+
+    Args:
+          bigtable_instance (str): Cloud Bigtable instance id to autoscale
+          high_cpu_threshold (float): If CPU is higher than this, scale up.
+          low_cpu_threshold (float): If CPU is higher than this, scale down.
+          short_sleep (int): How long to sleep after no operation
+          long_sleep (int): How long to sleep after the cluster nodes are
+                            changed
+    """
+    cluster_cpu = get_cpu_load()
+    print('Detected cpu of {}'.format(cluster_cpu))
+    if cluster_cpu > high_cpu_threshold:
+        scale_bigtable(bigtable_instance, True)
+        time.sleep(long_sleep)
+    elif cluster_cpu < low_cpu_threshold:
+        scale_bigtable(bigtable_instance, False)
+        time.sleep(short_sleep)
+    else:
+        print('CPU within threshold, sleeping.')
+        time.sleep(short_sleep)
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Scales Bigtable clusters based on CPU usage.')
+    parser.add_argument(
+        'bigtable_instance', help='ID of the Cloud Bigtable instance to '
+                                  'connect to.')
+    parser.add_argument(
+        '--high_cpu_threshold',
+        help='If Bigtable CPU usages is above this threshold, scale up',
+        default=0.6)
+    parser.add_argument(
+        '--low_cpu_threshold',
+        help='If Bigtable CPU usages is above this threshold, scale up',
+        default=0.2)
+    parser.add_argument(
+        '--short_sleep',
+        help='How long to sleep in seconds between checking metrics after no '
+             'scale operation',
+        default=60)
+    parser.add_argument(
+        '--long_sleep',
+        help='How long to sleep in seconds between checking metrics after a '
+             'scaling operation',
+        default=60 * 10)
+    args = parser.parse_args()
+
+    while True:
```

Comment:
I had it in main, but I moved it out here to make it easier to test main. Otherwise testing `while True` loops is annoying. Would probably have to make a bunch of changes just to test the code, this seemed like the simpler option.